osquery/remote/plugins/remote_enrollment.cpp

```diff
@@ -0,0 +1,94 @@
+#include <osquery/enrollment.h>
+
+#include "osquery/remote/requests.h"
+#include "osquery/remote/transports/http.h"
+#include "osquery/remote/serializers/json.h"
+
+#define MAX_TRIES 5
+
+namespace osquery {
+
+FLAG(string,
+     enrollment_endpoint,
+     "Not Specified",
+     "The endpoint for server side client enrollment");
+
+FLAG(string,
+     enrollment_port,
+     "Not Specified",
+     "The endpoint's port for server side client enrollment");
+FLAG(string,
+     enrollment_secret,
+     "Not Specified",
+     "The secret to be provided to the endpoint to faciliate enrollment");
+
+class RemoteEnrollmentPlugin : public EnrollmentPlugin {
+ public:
+  std::string getKey(const PluginRequest& request);
+
+ private:
+  Status enroll();
+  std::string client_key_ = "";
+};
+
+REGISTER(RemoteEnrollmentPlugin, "remote", "enrollment");
+
+Status RemoteEnrollmentPlugin::enroll() {
+  std::string location =
+      FLAGS_enrollment_endpoint + ":" + FLAGS_enrollment_port;
+  auto r = Request<HTTPTransport, JSONSerializer>(location);
```

Comment:
please include tests here to act as documentation for what you expect the server to return. see https://github.com/facebook/osquery/blob/master/osquery/remote/transports/http_tests.cpp for an example on how to create a dummy http server for use with tests.